R/logger.R

```diff
@@ -311,32 +311,47 @@ validate_log_level <- function(level) {
 }
 
 
+#' Make sure there are no unknown named arguments
+#' @keywords internal
+assert_no_unknown_args <- function() {
+    call <- match.call(definition = sys.function(1), call = sys.call(1), expand.dots = FALSE)
+    stopifnot("Unknown named argument provided." = is.null(names(call$`...`)))
```

Comment:
maybe better to supply the aegument(s)?

```suggestion
    if (!is.null(unknown_names <- names(call$`...`))) {
      stop("Unknown argument(s) provided: ", toString(unknown_names))
    }
```